app/flatpak-builtins-ls-remote.c

```diff
@@ -50,12 +51,33 @@ static GOptionEntry options[] = {
   { NULL }
 };
 
+typedef struct RemoteDirPair {
+  gchar *remote_name;
+  gchar *dir_id;
```

Comment:
I don't think using the dir_id for this works. For instance, the user directory has a NULL directory id, but when you later call flatpak_dir_get_system_by_id() with this you'll get back the system dir. Why not have a ref to the actual directory